python-sdk/nuscenes/eval/panoptic/panoptic_track_evaluator.py

```diff
@@ -0,0 +1,303 @@
+            }
+        """
+        # generate the areas for each unique instance in prediction.
+        unique_pred, counts_pred = np.unique(x_inst_in_cl[x_inst_in_cl > 0], return_counts=True)
+        id2idx_pred = {inst_id: idx for idx, inst_id in enumerate(unique_pred)}
+        matched_pred = np.array([False] * unique_pred.shape[0])
+
+        # generate the areas for each unique instance in ground truth.
+        unique_gt, counts_gt = np.unique(y_inst_in_cl[y_inst_in_cl > 0], return_counts=True)
+        id2idx_gt = {inst_id: idx for idx, inst_id in enumerate(unique_gt)}
+        matched_gt = np.array([False] * unique_gt.shape[0])
+
+        # generate intersection using offset
+        valid_combos = np.logical_and(x_inst_in_cl > 0, y_inst_in_cl > 0)
+        offset_combo = x_inst_in_cl[valid_combos] + self.offset * y_inst_in_cl[valid_combos]
+        unique_combo, counts_combo = np.unique(offset_combo, return_counts=True)
+
+        # Per-class accumulated stats
+        if scene is not None and cl < self.min_stuff_cls_id:
+            cl_preds = self.preds[scene]
+            cl_gts = self.gts[scene][cl]
+            cl_intersects = self.intersects[scene][cl]
+            self.update_dict_stat(cl_gts,
+                                  unique_gt[counts_gt > self.min_points],
+                                  counts_gt[counts_gt > self.min_points])
+            self.update_dict_stat(cl_preds,
+                                  unique_pred[counts_pred > self.min_points],
+                                  counts_pred[counts_pred > self.min_points])
+            valid_combos_min_point = np.zeros_like(y_inst_in_cl)  # instances which have more than self.min points
+            for valid_id in unique_gt[counts_gt > self.min_points]:
+                valid_combos_min_point = np.logical_or(valid_combos_min_point, y_inst_in_cl == valid_id)
+            y_inst_in_cl = y_inst_in_cl * valid_combos_min_point
+            valid_combos_ = np.logical_and(x_inst_row > 0, y_inst_in_cl > 0)
+            offset_combo_ = x_inst_row[valid_combos_] + self.offset * y_inst_in_cl[valid_combos_]
+            unique_combo_, counts_combo_ = np.unique(offset_combo_, return_counts=True)
+            self.update_dict_stat(cl_intersects, unique_combo_, counts_combo_)
+
+        # generate an intersection map, count the intersections with over 0.5 IoU as TP
+        gt_labels = unique_combo // self.offset
+        pred_labels = unique_combo % self.offset
+        gt_areas = np.array([counts_gt[id2idx_gt[g_id]] for g_id in gt_labels])
+        pred_areas = np.array([counts_pred[id2idx_pred[p_id]] for p_id in pred_labels])
+        intersections = counts_combo
+        unions = gt_areas + pred_areas - intersections
+        ious = intersections.astype(np.float32) / unions.astype(np.float32)
+
+        return counts_pred, counts_gt, gt_labels, pred_labels, matched_pred, matched_gt, id2idx_gt, id2idx_pred, ious
```

Comment:
good point,  the `match_gt/matched_pred` is expected to calculated outside, since `id2idx_pred/id2idx_gt` already contains the dimensions needed by `match_gt/matched_pred`. removed these 2 return variables.